AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -882,20 +900,26 @@ - (CGSize)collectionView:(UICollectionView *)collectionView layout:(UICollection
 
 - (UICollectionReusableView *)collectionView:(UICollectionView *)collectionView viewForSupplementaryElementOfKind:(NSString *)kind atIndexPath:(NSIndexPath *)indexPath
 {
+  if ([_registeredSupplementaryKinds containsObject:kind] == NO) {
+    [self registerSupplementaryNodeOfKind:kind];
+  }
+  
   UICollectionReusableView *view = nil;
   ASCellNode *node = [_dataController supplementaryNodeOfKind:kind atIndexPath:indexPath];
 
-  if (node.shouldUseUIKitCell && _asyncDataSourceFlags.interop) {
+  BOOL shouldDequeueExternally = _asyncDataSourceFlags.interopViewForSupplementaryElement && (_asyncDataSourceFlags.interopAlwaysDequeue || node.shouldUseUIKitCell);
```

Comment:
Great, this seems right -- though complicated, it satisfies some pretty tricky use cases.  It will be a happy day to find a further simplification to this...